acnportal/signals/tariffs/tou_tariff.py

```diff
@@ -0,0 +1,126 @@
+import json
+import os.path
+from datetime import datetime, timedelta
+from decimal import Decimal
+from copy import copy
+
+
+class TimeOfUseTariff(object):
+    """ Class to represent the time-of-use tariff schedule of a utility.
+
+    Args:
+        filename (str): Name of the tariff structure file excluding its extension which should be .json.
+        tariff_dir (str): Path to the directory where tariff files are stored. Defaults to the pre-installed tariff
+            directory included with the package.
+
+    Attributes:
+        name (str): Name of the tariff.
+        effective (str): Date when the tariff when into effect. Useful if a tariff has been modified but kept the same
+            name.
+    """
+    def __init__(self, filename, tariff_dir='tariff_schedules'):
+        with open(os.path.join(os.path.dirname(__file__), tariff_dir, filename + '.json')) as f:
+            tariff_file = json.load(f)
+        self.name = tariff_file['name']
+        self.effective = tariff_file['effective']
+        self._schedule = [TariffSchedule(s) for s in tariff_file['schedule']]
+        to_add = []
+        for s in self._schedule:
+            if s.end < s.start:
+                s_copy = copy(s)
+                s_copy.start = (1, 1)
+                to_add.append(s_copy)
+                s.end = (12, 31)
+        self._schedule.extend(to_add)
+        self._schedule.sort(key=lambda x: x.start)
+
+    def _get_tariff_schedule(self, dt):
+        """ Return the tariff schedule in effect for the given datetime.
+
+        Args:
+            dt (datetime): A datetime object.
+
+        Returns:
+            TariffSchedule: An object representing the tariff schedule for a given time.
+        """
+        valid_schedules = [s for s in self._schedule if s.dow_mask[dt.weekday()] and
+                           s.start <= (dt.month, dt.day) <= s.end]
+        if len(valid_schedules) == 0:
+            raise ValueError('No valid tariff schedule for {0}'.format(dt))
```

Comment:
Would this be better as a warning in case the user intended to have no tariff on certain days?

Reply:
I don't think so. I can't image a case where there is free energy for a day. Even if it were the case, its much better to force them to be specific since this is a way edge case and its more likely that the user made a mistake with defining the tariff schedule.